src/mds/CInode.cc

```diff
@@ -5015,9 +5021,6 @@ void CInode::validate_disk_state(CInode::validated_data *results,
 	    << "freshly-calculated rstats don't match existing ones (will be fixed)";
 	  in->mdcache->repair_inode_stats(in);
           results->raw_stats.repaired = true;
-          for (const auto &p : in->dirfrags){
-            in->mdcache->mds->damage_table.remove_dirfrag_damage_entry(p.second);
-          }
```

Comment:
Don't we fix some damage types with forward scrub? This removed block certainly suggests we do.

Do we need to be more fine-grained about what blocks is from doing IO?

Reply:
See the commit message in [mds: skip scrubbing damaged dirfrag](https://github.com/ceph/ceph/pull/61562/commits/7487845442b751a4a819243e96329cbda47b43bd) which I believe answers most of your question.

The kinds of damage we discover where we mark a dirfrag damaged has never been repaired by forward scrub.